### ISSUE TITLE
Errors when using reusing relative paths in output section

### ISSUE THREAD

**Issue by desilinguist** [EducationalTestingService/skll#252]
How to reproduce:

1. You want to put logs, models, predictions etc. in the same directory and so you use the same relative path to that directory in the config file.
2. You run the command `run_experiment <config_file>` from *outside* the directory where the config file lives. 

You will get the error:

```python
Traceback (most recent call last):
  File "/Users/nmadnani/anaconda/envs/garbage/bin/run_experiment", line 6, in <module>
    sys.exit(main())
  File "/Users/nmadnani/anaconda/envs/garbage/lib/python3.4/site-packages/skll/utilities/run_experiment.py", line 108, in main
    ablation=ablation, resume=args.resume)
  File "/Users/nmadnani/anaconda/envs/garbage/lib/python3.4/site-packages/skll/experiments.py", line 697, in run_configuration
    custom_learner_path) = _parse_config_file(config_file)
  File "/Users/nmadnani/anaconda/envs/garbage/lib/python3.4/site-packages/skll/config.py", line 448, in _parse_config_file
    os.makedirs(prediction_dir)
  File "/Users/nmadnani/anaconda/envs/garbage/lib/python3.4/os.py", line 237, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/Users/nmadnani/work/caaspp-wer-garbage-detection/skll/output'
```

This happens because when checking whether to create the directories in `_parse_config_file()` in `config.py`, we check the [relative path version](https://github.com/EducationalTestingService/skll/blob/master/skll/config.py#L452) instead of the properly located version relative to the config file.

**Comment by dan-blanchard** [comment]
Linking to #213, since it was a very similar issue.

**Comment by desilinguist** [comment]
Addressed by #287.
